crates/turbopath/src/anchored_system_path_buf.rs

```diff
@@ -0,0 +1,51 @@
+use std::path::{Path, PathBuf};
+
+use serde::Serialize;
+
+use crate::{AbsoluteSystemPathBuf, IntoSystem, PathValidationError};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default, Serialize)]
+pub struct AnchoredSystemPathBuf(PathBuf);
+
+impl TryFrom<&Path> for AnchoredSystemPathBuf {
+    type Error = PathValidationError;
+
+    fn try_from(path: &Path) -> Result<Self, Self::Error> {
+        if path.is_absolute() {
+            return Err(PathValidationError::NotRelative(path.to_path_buf()));
+        }
+
+        Ok(AnchoredSystemPathBuf(path.into_system()?))
+    }
+}
+
+impl AnchoredSystemPathBuf {
+    pub fn strip_root(
+        root: &AbsoluteSystemPathBuf,
+        path: &AbsoluteSystemPathBuf,
+    ) -> Result<Self, PathValidationError> {
+        let stripped_path = path
+            .as_path()
+            .strip_prefix(root.as_path())
+            .map_err(|_| PathValidationError::NotParent(root.to_string(), path.to_string()))?
+            .to_path_buf();
+
+        Ok(AnchoredSystemPathBuf(stripped_path))
+    }
+
+    pub fn new_unchecked(path: impl Into<PathBuf>) -> Self {
+        AnchoredSystemPathBuf(path.into())
+    }
+
+    pub fn as_path(&self) -> &Path {
+        self.0.as_path()
+    }
+
+    pub fn into_path_buf(self) -> PathBuf {
+        self.0
+    }
```

Comment:
Is there a reason for preferring this method instead `impl Into<PathBuf> for AnchoredSystemPathBuf`?